core/control-plane/contract/src/main/java/org/eclipse/dataspaceconnector/contract/offer/ContractOfferServiceImpl.java

```diff
@@ -57,8 +60,19 @@ public Stream<ContractOffer> queryContractOffers(ContractOfferQuery query, Range
         var agent = agentService.createFor(query.getClaimToken());
 
         return definitionService.definitionsFor(agent, range)
+                // ignore definitions with missing criteria (only if not SELECT_ALL)
+                .filter(definition -> (definition.getSelectorExpression().getCriteria() != null &&
```

Comment:
I think this filter does nothing, as when the `criteria` field is empty, then the `selectorExpression` equals to `SELECT_ALL`. Because `SELECT_ALL` means no criterion are provided.
(this pass:)
```java
    @Test
    void equality() {
        assertThat(AssetSelectorExpression.Builder.newInstance().build()).isEqualTo(AssetSelectorExpression.SELECT_ALL);
    }
```

Furthermore, the `selectorExpression` of the definition cannot be null.

Reply:
The `selectorExpression` cannot be null but the `criteriaList` can. There is no additional check in the builder class that validates whether the expresion is `SELECT_ALL` and otherwise the criteria list cannot be empty or null.

Reply:
> Because `SELECT_ALL` means no criterion are provided.

This should not be the case. I missed this test. `SELECT_ALL` means "select all". No criteria means "select none".